c_src/workitems.cc

```diff
@@ -487,19 +487,40 @@ RangeScanTask::RangeScanTask(ErlNifEnv * caller_env,
                              RangeScanOptions & options,
                              SyncObject * sync_obj)
 : WorkTask(caller_env, caller_ref, db_handle),
-    options_(options),
-    start_key_(start_key),
-    has_end_key_(bool(end_key)),
-    sync_obj_(sync_obj)
+  options_(options),
+  start_key_(start_key),
+  has_end_key_(bool(end_key)),
+  sync_obj_(sync_obj),
+  range_filter_(0),
+  extractor_(0)
 {
     if (end_key) {
         end_key_ = *end_key;
     }
+
+    if(options_.useRangeFilter_) {
+        if(options_.encodingType_ == Encoding::MSGPACK)
+            extractor_ = new ExtractorMsgpack();
+        else {
+            ThrowRuntimeError("An invalid object encoding was specified");
```

Comment:
streaming_start() in eleveldb.cc calls "new" operator on this object without try/catch.  Riak will crash if this throws.